PWGHF/D2H/Tasks/taskLc.cxx

```diff
@@ -305,12 +363,43 @@ struct HfTaskLc {
       registry.fill(HIST("Data/hImpParErrProng1"), candidate.errorImpactParameter1(), pt);
       registry.fill(HIST("Data/hImpParErrProng2"), candidate.errorImpactParameter2(), pt);
       registry.fill(HIST("Data/hDecLenErr"), candidate.errorDecayLength(), pt);
+
+      if constexpr (applyMl) {
+        double Mass(-1);
+        double outputBkg(-1);
+        double outputFD(-1);
+        if (candidate.isSelLcToPKPi() >= selectionFlagLc) {
+          Mass = hfHelper.invMassLcToPKPi(candidate);
+          outputBkg = candidate.mlProbLcToPKPi()[0]; /// bkg score
+          outputFD = candidate.mlProbLcToPKPi()[1];  /// non-prompt score
+          /// Fill the ML outputScores and variables of candidate
+          registry.get<THnSparse>(HIST("hBdtScoreVsVars"))->Fill(Mass, pt, ptProng0, ptProng1, ptProng2, nTracks, chi2PCA, decayLength, decayLengthXY, cpa, cpaXY, outputBkg, outputFD, 0);
+        } else if (candidate.isSelLcToPiKP() >= selectionFlagLc) {
+          Mass = hfHelper.invMassLcToPiKP(candidate);
+          outputBkg = candidate.mlProbLcToPKPi()[0]; /// bkg score
+          outputFD = candidate.mlProbLcToPKPi()[1];  /// non-prompt score
```

Comment:
I guess that here you should use `mlProbLcToPiKP`
Also, typically we put the FD score in 3rd position (this is not mandatory, but it is what we usually do, so it would be good to have the same convention everywhere in the code)
Also same comment about possibly empty vectors (see above)